metaci/release/models.py

```diff
@@ -123,37 +143,21 @@ def update_from_github(self):
 
     def save(self, *args, **kw):
         super().save(*args, **kw)
-        self.create_default_implementation_step(
-            "release_deploy",
-            get_default_sandbox_date(),
-            get_default_sandbox_date(),
-            datetime.time(8),
-            datetime.time(18),
-        )
-        self.create_default_implementation_step(
-            "release",
-            get_default_sandbox_date(),
-            get_default_sandbox_date(),
-            datetime.time(8),
-            datetime.time(18),
-        )
-        self.create_default_implementation_step(
-            "push_sandbox",
-            get_default_sandbox_date(),
-            get_default_sandbox_date(),
-            datetime.time(18),
-            datetime.time(23, 59),
-        )  # time will vary depending on product
-        self.create_default_implementation_step(
-            "push_production",
-            get_default_production_date(),
-            get_default_production_date(),
-            datetime.time(18),
-            datetime.time(23, 59),
-        )  # time will vary depending on product
+        for step_dict in self.repo.default_implementation_steps:
+            step_info = DefaultImplementationStep(**step_dict)
+            start = step_info.start(self, step_dict)
+            end = step_info.end(start, step_dict)
+            self.create_default_implementation_step(
+                step_dict["role"],
+                start,
+                end,
+            )
```

Comment:
You could just pass `step_info` here, and let `create_default_implementation_step` retrieve values from it.